File: backend/app/schemas/wishlist.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime, date

class WishlistCreate(BaseModel):
    accommodation_id: str
    desired_date: Optional[date] = None
    notify_enabled: bool = True
    notification_type: Optional[str] = None  # ios_webview, android_fcm, kakao
    fcm_token: Optional[str] = None

class WishlistUpdate(BaseModel):
    desired_date: Optional[date] = None
    notify_enabled: Optional[bool] = None
    notification_type: Optional[str] = None
    fcm_token: Optional[str] = None

class WishlistResponse(BaseModel):
    id: str
    user_id: str
    accommodation_id: str
    desired_date: Optional[date] = None
    is_active: bool
    notify_enabled: bool
    notification_type: Optional[str] = None
    fcm_token: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/wishlist.ts
export interface Wishlist {
  id: string;
  user_id: string;
  accommodation_id: string;
  desired_date?: string; // ISO date string (YYYY-MM-DD)
  is_active: boolean;
  notify_enabled: boolean;
  notification_type?: "ios_webview" | "android_fcm" | "kakao";
  fcm_token?: string;
  created_at: string;
  updated_at: string;
}

export interface WishlistCreate {
  accommodation_id: string;
  desired_date?: string; // ISO date string (YYYY-MM-DD)
  notify_enabled?: boolean;
  notification_type?: "ios_webview" | "android_fcm" | "kakao";
  fcm_token?: string;
}

export interface WishlistUpdate {
  desired_date?: string;
  notify_enabled?: boolean;
  notification_type?: "ios_webview" | "android_fcm" | "kakao";
  fcm_token?: string;
}

export interface WishlistWithAccommodation extends Wishlist {
  accommodation?: {
    id: string;
    name: string;
    region: string;
    first_image?: string;
    avg_score?: number;
  };
}
